lib/train/extras/os_detect_arista_eos.rb

```diff
@@ -0,0 +1,30 @@
+# encoding: utf-8
+# author: Jere Julian
+#
+# Arista EOS has 2 modes. Most compliance tests will use the network CLI
+# but when working with vagrant, its common to encounter the raw bash shell.
+require 'json'
+
+module Train::Extras
+  module DetectAristaEos
+    def detect_arista_eos
+      if unix_file?('/usr/bin/FastCli')
+        output = @backend.run_command('FastCli -p 15 -c "show version | json"').stdout
+        @platform[:name] = 'arista_eos_bash'
+        family = 'fedora'
+      else
+        output = @backend.run_command('show version | json').stdout
+      end
+
+      unless output.empty?
+        eos_ver = JSON.parse(output)
+        @platform[:name] = @platform[:name] || 'arista_eos'
+        family ||= 'arista_eos'
```

Comment:
When you are in the bash shell, then you can do all the fedora things.  However, when you login the way you would to a  typical running switch, you are in a custom EOS CLI.  You can still prefix linux commands with 'bash' from this mode but common commands are things like 'show running-config', 'show interfaces status', etc.  Thus, things that expect to be in a regular linux shell will need special handling.

I'm definitely willing to go another route, if appropriate.

You can test yourself by registering at http://Arista.com/, then going to Software Downloads --> vEOS and downloading the files for your preferred hypervisor.